### PR TITLE
Remove coming-soon redirect on FAQ pages and add acts_like_production method for staging & production

### PR DESCRIPTION
**Pull request by embarnard**

### CODE REVIEW

File: app/controllers/state_file/state_file_pages_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def redirect_locale_home
     end
 
     def fake_direct_file_transfer_page
-      return render "public_pages/page_not_found", status: 404 if Rails.env.production? || Rails.env.staging?
+      return render "public_pages/page_not_found", status: 404 if Rails.env.production?
```
Comment (tofarr) [inline app/controllers/state_file/state_file_pages_controller.rb:11]: Should this be `acts_like_production?`

File: app/services/irs_api_service.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def self.df_return_sample
   end
 
   def self.import_federal_data(authorization_code, state_code)
-    unless Rails.env.production? || Rails.env.staging?
+    unless Rails.env.production?
```
Comment (tofarr) [inline app/services/irs_api_service.rb:14]: Again - should it be `acts_like_production?`

File: app/views/state_file/questions/initiate_data_transfer/_data_transfer_buttons.html.erb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 <% end %>
 
 <% if @irs_df_transfer_link.present? %>
-  <% if Rails.env.production? || Rails.env.staging? %>
+  <% if Rails.env.production? %>
```
Comment (tofarr) [inline app/views/state_file/questions/initiate_data_transfer/_data_transfer_buttons.html.erb:9]: Should this be `acts_like_production?`

File: app/controllers/state_file/state_file_pages_controller.rb (hunk 1 above)
Comment (embarnard) [reply app/controllers/state_file/state_file_pages_controller.rb:11]: I thought so too at first but they will not have access to the IRS testing site so will have to go through fake-df flow more context here: https://cfastaff.slack.com/archives/C0544ERAFQV/p1706825778050309

File: app/services/irs_api_service.rb (hunk 2 above)
Comment (embarnard) [reply app/services/irs_api_service.rb:14]: also same as above

File: app/views/state_file/questions/initiate_data_transfer/_data_transfer_buttons.html.erb (hunk 3 above)
Comment (embarnard) [reply app/views/state_file/questions/initiate_data_transfer/_data_transfer_buttons.html.erb:9]: also same as above

**Review by tofarr — APPROVED** [review]

**Pull request merged**